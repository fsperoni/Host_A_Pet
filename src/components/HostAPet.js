import React, { useState, useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import jwt from "jsonwebtoken";
import Header from "./Header";
import AllRoutes from "./AllRoutes";
import useLocalStorage from "./useLocalStorage";
import HostAPetApi from "./Api";
import UserContext from "./UserContext";
import Footer from "./Footer";

export const TOKEN_ID = "HostAPet-token";

/**
 * Main app component that holds all the states, AJAX calls and renders 
 * the different routes as applicable.
 */
const HostAPet = () => {
  const [infoLoaded, setInfoLoaded] = useState(false);
  const [currentUser, setCurrentUser] = useState(null);
  const [token, setToken] = useLocalStorage(TOKEN_ID);

  useEffect(function loadUserInfo() {

    const getCurrentUser = async() => {
      if (token) {
        try {
          const { username } = jwt.decode(token);
          HostAPetApi.token = token;
          const currentUser = await HostAPetApi.getCurrentUser(username);
          setCurrentUser(currentUser);
        } catch (err) {
          setCurrentUser(null);
        }
      }
      setInfoLoaded(true);
    }
    setInfoLoaded(false);
    getCurrentUser();
  }, [token]);

  const logout = () => {
    setCurrentUser(null);
    setToken(null);
  }
  const signup = async(signupData) => {
    try {
      const token = await HostAPetApi.signup(signupData);
      setToken(token);
      return { success: true };
    } catch (errors) {
      return { success: false, errors };
    }
  }

  const login = async(loginData) => {
    try {
      const token = await HostAPetApi.login(loginData);
      setToken(token);
      return { success: true };
    } catch (errors) {
      return { success: false, errors };
    }
  }
  return (
    <BrowserRouter>
      <UserContext.Provider
          value={{ currentUser, setCurrentUser }}>
        <main className="flex-shrink-0">
        <Header logout={logout} />
          <AllRoutes logout={logout} login={login} signup={signup} />
        </main>
        <Footer />
      </UserContext.Provider>
    </BrowserRouter>
  );
}

export default HostAPet;