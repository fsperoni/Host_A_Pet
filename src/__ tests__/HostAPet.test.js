import { render } from "@testing-library/react";
import HostAPet from "../components/HostAPet";

// smoke test
it('renders without crashing', function() {
  render(<HostAPet />);

});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<HostAPet />);
  expect(asFragment()).toMatchSnapshot();
});
