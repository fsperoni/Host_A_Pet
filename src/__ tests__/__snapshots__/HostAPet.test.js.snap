// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="flex-shrink-0"
  >
    <nav
      class="navbar navbar-expand-sm"
      direction="horizontal"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          <div
            class="d-flex"
          >
            <img
              alt="Host a Pet"
              src="logo.png"
            />
            <h1
              class="my-auto"
            >
              Host a Pet
            </h1>
          </div>
        </a>
        <ul
          class="ms-5 navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="mr-2 active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="mr-2"
              href="/newBooking"
            >
              New Booking
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="mr-2"
              href="/myBookings"
            >
              My Bookings
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="mr-2"
              href="/myAvailabilities"
            >
              My Availabilities
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="mr-2"
              href="/myPets"
            >
              My Pets
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/myProfile"
            >
              My Profile
            </a>
          </li>
        </ul>
        <div
          class="login"
        >
          <a
            class=""
            href="/login"
          >
            Login
          </a>
          <a
            class="ms-3"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
    <div
      class="container mt-3"
    >
      <h1
        class="text-center"
      >
        Host a Pet!
      </h1>
      <p
        class="text-center"
      >
        We connect pet owners and pet hosts for amazing experiences. Sign up today and either search for a pet to host or someone to host your pet. You can browse by distance from your postal code, or filter results by host and pet owner ratings.
      </p>
      <p>
        Review hostings, assigning a score, or go through previous hostings feedback to help you choose the best host for your pet, or the best pet for you to host.
      </p>
      <img
        class="img-fluid rounded mx-auto d-block"
        src="pets.jpeg"
      />
    </div>
  </main>
  <footer
    class="footer mt-auto py-3"
  >
    <p
      class="text-center my-auto"
    >
      Â© Fabio Speroni 2022
    </p>
  </footer>
</DocumentFragment>
`;
